iModelJsNodeAddon/IModelJsNative.cpp

```diff
@@ -4658,6 +4681,33 @@ struct NativeECSqlStatement : BeObjectWrap<NativeECSqlStatement> {
         return Napi::String::New(Env(), m_stmt.GetNativeSql());
     }
 
+    static NativeECSqlRowArg getECSqlRowArg(NapiInfoCR info) {
+        REQUIRE_ARGUMENT_ANY_OBJ(0, optObj);
+        BeJsValue opts(optObj);
+        if (!opts.isBoolMember("abbreviateBlobs"))
+            BeNapi::ThrowJsException(info.Env(), "abbreviateBlobs argument missing");
+        if (!opts.isBoolMember("convertClassIdsToClassNames"))
+            BeNapi::ThrowJsException(info.Env(), "convertClassIdsToClassNames argument missing");
+        if (!opts.isNumericMember("rowFormat"))
+            BeNapi::ThrowJsException(info.Env(), "rowFormat argument missing");
+        if (!opts.isBoolMember("includeMetaData"))
+            BeNapi::ThrowJsException(info.Env(), "includeMetaData argument missing");
+
+        bool useJsName = (NativeQueryRowFormat)opts["rowFormat"].asInt() == NativeQueryRowFormat::UseJsPropertyNames;

```

Comment:
Is it best to use `QueryRowFormat` as an parameter to this `ToRow` function if we can only check for UseJsPropertyNames flag? It can seem misleading, that the other two options are also available. Why not have a simple boolean instead?

Reply:
I talked with Akshay and it make sense to just have one arg for getRow() on typescript side. Though internally it will set these flag accordingly. But we do not have to expose internal flags to users.